scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,272 @@
+
+
+#### Compare `DropletUtils` to `miQC` filtering
+
+We may want to compare certain aspects of cells that were filtered out with DropletUtils vs miQC.
+Chunk below, currently pending content..., can do that!
+
+```{r}
+
+```
+
+### Filter ADT
+
+This data does have a control antibody (IgG1), so we can leverage this ADT for QC and filtering.
+```{r}
+# Filter ADT
+# http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters
+
+controls <- grep("^Ig", rownames(altExp(filtered_sce)))
+adt_qc_df <- DropletUtils::cleanTagCounts(altExp(filtered_sce), controls=controls) # I feel like this control might be crap!
+
+summary(adt_qc_df$high.controls) # these are all NA????????
+summary(adt_qc_df$discard) # these are MOSTLY NA????????
+
+# Let's do this - remove cells that have zero ambient concentrations:
+# This removes 11 cells
+cells_to_retain <- which(adt_qc_df$zero.ambient == FALSE)
+filtered_sce <- filtered_sce[, cells_to_retain]
+dim(filtered_sce)
+```
+
+
+
+### Normalize RNA and ADT:
+
+
+```{r}
+# Normalize RNA
+qclust <- scran::quickCluster(filtered_sce)
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
+norm_sce <- scater::logNormCounts(filtered_sce)
+
+# Normalize ADT
+
+# Calculate baseline:
+# Try out control-based normalization here 
+#  http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#control-based-normalization
+control_sf <- scater::librarySizeFactors(altExp(norm_sce), subset_row=controls)  # `controls` was previously defined during filtering
+sum(control_sf == 0) # _HOLY WOW_
+
+# Go for median instead, BUT TODO: would we want to exclude control here during baseline calculation? 
+baseline <- DropletUtils::ambientProfileBimodal(altExp(filtered_sce)) # !! currently this does not exclude controls
+median_sf <- scater::medianSizeFactors(altExp(norm_sce), reference = baseline)
+sum(median_sf == 0) # none are 0 here, soooo let's do this?
+
+# Normalize with the median size factors
+altExp(norm_sce) <- scater::logNormCounts(altExp(norm_sce), 
+                                          size.factors = median_sf)
+
+```
+
+### Behold, the filtered and normalized SCE:
+
+```{r}
+norm_sce
+
+altExp(norm_sce)
+```
+
+## Dimension reduction
+
+HVGs:
+```{r}
+num_genes <- 5000 # eh?
+
+gene_variance <- scran::modelGeneVar(norm_sce)
+hvg_list <- scran::getTopHVGs(gene_variance,
+                               n = num_genes)
+```
+
+PCA:
+```{r}
+# Do we want to set the number of PCs?
+norm_sce <- scater::runPCA(norm_sce,
+                           subset_row = hvg_list)
+```
+
+UMAP:
+```{r I map you map we all map for MapMap}
+# We can arbitrarily cluster first just to have some colors in there for funsies
+k <- 10
+
+# perform the clustering
+kclusters <- clusterRows(
+  reducedDim(norm_sce, "PCA"),
+  KmeansParam(centers = k)
+)
```

Comment:
I'm not sure we need to do clustering at this stage, but if we do, we should do a graph method, as that is much more standard & probably better for these data. On second thought, probably it is worth doing, then we can wave at the cluster/cell type correspondences (or lack therof).